From e766089c7395d40433a2a406521238959e31eb93 Mon Sep 17 00:00:00 2001
From: stephen hemminger <stephen@networkplumber.org>
Date: Fri, 22 Feb 2013 08:01:10 +0000
Subject: [PATCH 602/858] bnx2x: remove dead code and make local funcs static

Sparse warned about several functions that were unnecessarily global.
After making them static, discovered that several functions were actually never used.

Compile tested only.

Signed-off-by: Stephen Hemminger <stephen@networkplumber.org>
Acked-by: Eilon Greenstein <eilong@broadcomo.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 732ac8cacbbbdbf9f5ba92299cc747db3abfefd0)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c | 58 ------------------------
 drivers/net/ethernet/broadcom/bnx2x/bnx2x_vfpf.c |  4 +-
 2 files changed, 2 insertions(+), 60 deletions(-)

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c
index a4cd1f4..bafe2c2 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c
@@ -13013,64 +13013,6 @@ static int bnx2x_84833_common_init_phy(struct bnx2x *bp,
  return 0;
 }
 
-static int bnx2x_84833_pre_init_phy(struct bnx2x *bp,
-        struct bnx2x_phy *phy,
-        u8 port)
-{
- u16 val, cnt;
- /* Wait for FW completing its initialization. */
- for (cnt = 0; cnt < 1500; cnt++) {
-  bnx2x_cl45_read(bp, phy,
-    MDIO_PMA_DEVAD,
-    MDIO_PMA_REG_CTRL, &val);
-  if (!(val & (1<<15)))
-   break;
-  usleep_range(1000, 2000);
- }
- if (cnt >= 1500) {
-  DP(NETIF_MSG_LINK, "84833 reset timeout\n");
-  return -EINVAL;
- }
-
- /* Put the port in super isolate mode. */
- bnx2x_cl45_read(bp, phy,
-   MDIO_CTL_DEVAD,
-   MDIO_84833_TOP_CFG_XGPHY_STRAP1, &val);
- val |= MDIO_84833_SUPER_ISOLATE;
- bnx2x_cl45_write(bp, phy,
-    MDIO_CTL_DEVAD,
-    MDIO_84833_TOP_CFG_XGPHY_STRAP1, val);
-
- /* Save spirom version */
- bnx2x_save_848xx_spirom_version(phy, bp, port);
- return 0;
-}
-
-int bnx2x_pre_init_phy(struct bnx2x *bp,
-      u32 shmem_base,
-      u32 shmem2_base,
-      u32 chip_id,
-      u8 port)
-{
- int rc = 0;
- struct bnx2x_phy phy;
- if (bnx2x_populate_phy(bp, EXT_PHY1, shmem_base, shmem2_base,
-          port, &phy) != 0) {
-  DP(NETIF_MSG_LINK, "populate_phy failed\n");
-  return -EINVAL;
- }
- bnx2x_set_mdio_clk(bp, chip_id, phy.mdio_ctrl);
- switch (phy.type) {
- case PORT_HW_CFG_XGXS_EXT_PHY_TYPE_BCM84833:
- case PORT_HW_CFG_XGXS_EXT_PHY_TYPE_BCM84834:
-  rc = bnx2x_84833_pre_init_phy(bp, &phy, port);
-  break;
- default:
-  break;
- }
- return rc;
-}
-
 static int bnx2x_ext_phy_common_init(struct bnx2x *bp, u32 shmem_base_path[],
          u32 shmem2_base_path[], u8 phy_index,
          u32 ext_phy_type, u32 chip_id)
diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_vfpf.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_vfpf.c
index 1cbc04c..d463152 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_vfpf.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_vfpf.c
@@ -98,7 +98,7 @@ static inline int bnx2x_pfvf_status_codes(int rc)
  }
 }
 
-int bnx2x_send_msg2pf(struct bnx2x *bp, u8 *done, dma_addr_t msg_mapping)
+static int bnx2x_send_msg2pf(struct bnx2x *bp, u8 *done, dma_addr_t msg_mapping)
 {
  struct cstorm_vf_zone_data __iomem *zone_data =
   REG_ADDR(bp, PXP_VF_ADDR_CSDM_GLOBAL_START);
@@ -141,7 +141,7 @@ int bnx2x_send_msg2pf(struct bnx2x *bp, u8 *done, dma_addr_t msg_mapping)
  return 0;
 }
 
-int bnx2x_get_vf_id(struct bnx2x *bp, u32 *vf_id)
+static int bnx2x_get_vf_id(struct bnx2x *bp, u32 *vf_id)
 {
  u32 me_reg;
  int tout = 10, interval = 100; /* Wait for 1 sec */
-- 
1.8.3.1